src/SprykerEco/Zed/Unzer/Business/Credentials/UnzerCredentialsCreator.php

```diff
@@ -73,9 +76,16 @@ public function __construct(
      */
     public function createUnzerCredentials(UnzerCredentialsTransfer $unzerCredentialsTransfer): UnzerCredentialsResponseTransfer
     {
-        return $this->getTransactionHandler()->handleTransaction(function () use ($unzerCredentialsTransfer) {
-            return $this->executeCreateUnzerCredentials($unzerCredentialsTransfer);
-        });
+        $propelConnection = Propel::getConnection();
+        $propelConnection->beginTransaction();
+        $unzerCredentialsResponseTransfer = $this->executeCreateUnzerCredentials($unzerCredentialsTransfer);
+        if (!$unzerCredentialsResponseTransfer->getIsSuccessful()) {
+            $propelConnection->rollBack();
+        } else {
+            $propelConnection->commit();
+        }
```

Comment:
It's more a matter of taste, but usually, we are recommended to use an early return instead of `else`:

```suggestion
        if (!$unzerCredentialsResponseTransfer->getIsSuccessful()) {
            $propelConnection->rollBack();

            return $unzerCredentialsResponseTransfer;
        }

        $propelConnection->commit();
        
        return $unzerCredentialsResponseTransfer;
```